lib/gemini.js

```diff
@@ -23,39 +26,49 @@ function requireWithNoCache(moduleName) {
     return result;
 }
 
-function Gemini(config, overrides) {
-    config = config || DEFAULT_CFG_NAME;
-    this.config = new Config(config, overrides);
-    var _this = this;
+module.exports = inherit(EventEmitter, {
+    __constructor: function(config, overrides) {
+      config = config || DEFAULT_CFG_NAME;
```

Comment:
Found something called [chained-emitter](https://www.npmjs.com/package/chained-emitter) which seems to do the trick:
`var ChainedEmitter = require('chained-emitter').EventEmitter`
then, we can do this:
```
 return _this.emit('startRunner', runnerInstance)
              .then(function() {
                  return _this.readTests(paths);
              })
              .then(function(rootSuite) {
                  options.reporters.forEach(applyReporter.bind(null, runnerInstance));
                  return runnerInstance.run(rootSuite);
              })
              .then(function(data) {
                  _this.emit('endRunner', runnerInstance, data);
                  return data;
              });
```
 What still bothers me though a bit is the order of the promises.. I would like to emit the 'startRunner' after `_this.readTests(paths)` but then we have to rely that the plugins event handler passes through the rootSuite to the next step and making that assumption just doesn't feel right. The same issue is with the 'endRunner' emitter, the 'data' would need to passed through it. Any thoughts on this?

From the plugins perspective, it looks like this currently:
```
gemini.on('startRunner', function(runner) {
    var deferred = Q.defer();

    server.start(opts, function(rootUrl) {
      gemini.config.rootUrl = rootUrl;
      deferred.resolve();
    });

    return deferred.promise;
  });
```